### ISSUE TITLE
Build error

### ISSUE THREAD

**Issue by YLavrov13** [m-stein/linux#2]
I'm trying to compile Genode for usbarmory. After executing this command **make ARCH=arm zImage LOADADDR=0x80008000 modules** from the root dir of the downloaded linux kernel, it gives me the following error.

Makefile:664: Cannot use CONFIG_CC_STACKPROTECTOR_STRONG: -fstack-protector-strong not supported by compiler
  CHK     include/config/kernel.release
  CHK     include/generated/uapi/linux/version.h
  CHK     include/generated/utsrelease.h
make[1]: 'include/generated/mach-types.h' is up to date.
  CC      kernel/bounds.s
gcc: error: unrecognized argument in option ‘-mabi=apcs-gnu’
gcc: note: valid arguments to ‘-mabi=’ are: ms sysv
gcc: error: unrecognized command line option ‘-mlittle-endian’
gcc: error: unrecognized command line option ‘-mapcs’
gcc: error: unrecognized command line option ‘-mno-sched-prolog’
Kbuild:35: recipe for target 'kernel/bounds.s' failed
make[1]: **\* [kernel/bounds.s] Error 1
Makefile:980: recipe for target 'prepare0' failed
make: **\* [prepare0] Error 2

Please advise

**Comment by m-stein** [comment]
Hey Yevgeny,

If you only want to try the default setup, you can simply use the run
script [1] by doing 'make run/tz_vmm' in your build directory. The
script automatically downloads a precompiled Linux image. Have you tried
that?

If you're really interested in customizing your Normal World Linux,
there is also a link in the run script to a detailed explanation on how
to compile the images [2]. I assume that your approach fails because of
the wrong toolchain. Does the tutorial fix your problem?

Cheers,
Martin

[1] https://github.com/genodelabs/genode/blob/master/repos/os/run/tz_vmm.run

[2]
https://github.com/m-stein/genode_binaries/blob/master/tz_vmm/usb_armory/README

Am 05.04.2016 um 21:49 schrieb Yevgeny:

**Comment by YLavrov13** [comment]
Yes, I've already tried: make run/tz_vmm, however, it doesn't create the **uImage** in the **var/run/tz_vmm/** directory.

I started off with this tutorial: https://github.com/inversepath/usbarmory/wiki/Genode-OS and everything went fine untill this point: cp var/run/tz_vmm/uImage to SD_CARD. Err: no such file or directory"

I've taken the following steps provided in the first link of you msg.
export TARGET_DEV=/dev/sdX
cd genode-16.02
./tool/create_builddir hw_usb_armory
./tool/create_uboot hw_usb_armory
sudo dd if=contrib/uboot/build/hw_usb_armory/mmc_img of=$TARGET_DEV bs=1K seek=1 conv=fsync
cd genode-16.02/build/hw_usb_armory/
echo "RUN_OPT += --include image/uboot" >> etc/build.conf
make run/tz_vmm

and ended up with the following error:

couldn't execute "mkimage": no such file or directory
    while executing
"exec mkimage -A arm -O linux -T kernel -C $compress_type -a $load_addr  -e $entrypoint -d $bin_img$bin_ext $uboot_img"
    (procedure "run_image" line 23)
    invoked from within
"run_image $elf_img"
    (procedure "run_boot_dir" line 136)
    invoked from within
"run_boot_dir $binaries"
    (procedure "build_boot_image" line 2)
    invoked from within
"build_boot_image [join $boot_modules " "]"
    (file "/home/yevgeny/Desktop/genode-16.02/repos/os/run/tz_vmm.run" line 265)
    invoked from within
"source $include_name"
    ("foreach" body line 6)
    invoked from within
"foreach include_name [get_cmd_arg --include ""] {
    # first check if the include name is absolute
    if {[string first "/" $include_name] == 0} {
        puts ..."

if instead of echo "RUN_OPT += --include image/uboot" >> etc/build.conf I change RUN_OPT = --include boot_dir/hw to RUN_OPT = --include boot_dir/hw image/uboot the build succeeds but no uImage gets created.

I'm using clean Ubuntu 15.10 VM, Genode15.05 x86-64 tool chain, and the latest version of Genode 16.02

**Comment by m-stein** [comment]
Hi Yevgeny,

I would like to move this discussion to the Genode mailing list at this
point.

Do you agree with that?

Cheers,
Martin

Am 17.04.2016 um 09:09 schrieb Yevgeny:

**Comment by YLavrov13** [comment]
Yes, of course

**Comment by m-stein** [comment]
Ok, I have answered the mail on the mailing list. I have changed the
subject text to "Re: [tz_vmm] Build error (#2)".

Am 20.04.2016 um 18:58 schrieb Yevgeny:

**Comment by mbwmbw1337** [comment]
Try
`make ARCH="arm64"`

**Comment by mbwmbw1337** [comment]
> I'm trying to compile Genode for usbarmory. After executing this command **make ARCH=arm zImage LOADADDR=0x80008000 modules** from the root dir of the downloaded linux kernel, it gives me the following error.
> 
> Makefile:664: Cannot use CONFIG_CC_STACKPROTECTOR_STRONG: -fstack-protector-strong not supported by compiler
> CHK include/config/kernel.release
> CHK include/generated/uapi/linux/version.h
> CHK include/generated/utsrelease.h
> make[1]: 'include/generated/mach-types.h' is up to date.
> CC kernel/bounds.s
> gcc: error: unrecognized argument in option ‘-mabi=apcs-gnu’
> gcc: note: valid arguments to ‘-mabi=’ are: ms sysv
> gcc: error: unrecognized command line option ‘-mlittle-endian’
> gcc: error: unrecognized command line option ‘-mapcs’
> gcc: error: unrecognized command line option ‘-mno-sched-prolog’
> Kbuild:35: recipe for target 'kernel/bounds.s' failed
> make[1]: *** [kernel/bounds.s] Error 1
> Makefile:980: recipe for target 'prepare0' failed
> make: *** [prepare0] Error 2
> 
> Please advise

Try
`make ARCH="arm64"`
